avocado/core/nrunner.py

```diff
@@ -130,6 +130,52 @@ def __repr__(self):
                           self.kwargs, self.tags, self.requirements,
                           self.variant)
 
+    @property
+    def identifier(self):
+        """Runnable identifier respecting user's format string.
+
+        This is still experimental and we have room for improvements.
+
+        This property it will return an unique identifier for this runnable.
+        Please use this property in order to respect user's customization.
+
+        By default runnables has its '{uri}' as identifier.
+
+        Custom formatter can be configured and currently we accept the
+        following values as normal f-strings replacements: {uri}, {args},
+        and {kwargs}. "args" and "kwargs" are special cases.
+
+        For args, since it is a list, you can use in two different ways:
+        "{args}" for the entire list, or "{args[n]}" for a specific element
+        inside this list.  The same is valid when using "{kwargs}". With
+        kwargs, since it is a dictionary, you have to specify a key as index
+        and then the values are used. For instance if you have a kwargs value
+        named 'DEBUG', a valid usage could be: "{kwargs[DEBUG]}" and this will
+        print the current value to this variable (i.e: True or False).
+
+        Since this is formatter, combined values can be used. Example:
+        "{uri}-{args}".
+        """
+        fmt = self.config.get("runner.identifier_fmt")
```

Comment:
This is part of a much longer discussion that I intended to revisit and write a better explanation about it.... but, let me try to add some early context: IMO we need to better define what controls a runnable's behavior, in an explicit way. Having Avocado's entire configuration is not the right granularity IMO, and it makes it virtually IMPOSSIBLE to "replay" a runnable by executing `avocado-runner-$kind runnable-run`, which IMO is a nice check for the simplicity of the overall solution.

But, until that is better explained, and discussed, and hopefully improved, this LGTM.